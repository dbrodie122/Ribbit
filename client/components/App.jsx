import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Home from './homepage/Home';
import Room from './room/Room';
import Login from './homepage/Login';
import Profile from './profile/Profile';

const App = props => (
  <div>
    <Navbar />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/login" component={Login} />
      <Route exact path="/profile" component={Profile} />
      <Route path={`/:${props.roomName}`} history={props.history} component={Room} />
    </Switch>
  </div>
);

export default App;
