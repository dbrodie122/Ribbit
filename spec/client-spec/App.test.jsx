import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../../client/components/App';
import { Provider } from 'react-redux';
import store from './../../client/store';

test('App component snapshot test', () => {
  const component = renderer.create(<Provider store={store}><Router><App /></Router></Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
