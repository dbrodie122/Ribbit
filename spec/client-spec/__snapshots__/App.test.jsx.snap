// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <header
      className="header"
    >
      <div
        className="navbar navbar-default navbar-fixed-top"
        role="navigation"
      >
        <div
          className="container"
        >
          <div
            className="navbar-header"
          >
            <a
              className="navbar-brand scroll-to"
              href="/"
              onClick={[Function]}
            >
              <img
                alt="logo"
                className="hidden-xs hidden-sm"
                src="../ribbit.png"
              />
              <span
                className="sr-only"
              >
                Go to homepage
              </span>
            </a>
            <div
              className="navbar-buttons"
            >
              <button
                className="navbar-toggle navbar-btn"
                data-target=".navbar-collapse"
                data-toggle="collapse"
                type="button"
              >
                Menu
                <i
                  className="pe-7s-menu"
                />
              </button>
            </div>
          </div>
          <div
            className="collapse navbar-collapse navbar-right"
            id="navigation"
          >
            <ul
              className="nav navbar-nav"
            >
              <li>
                <a
                  className="scroll-to"
                  href="#features"
                >
                  Features
                </a>
              </li>
              <li>
                <a
                  className="scroll-to"
                  href="#integrations"
                >
                  Integrations
                </a>
              </li>
              <li>
                <a
                  className="scroll-to"
                  href="#testimonials"
                >
                  Testimonials
                </a>
              </li>
              <li>
                <a
                  className="scroll-to"
                  href="#contact"
                >
                  Contact
                </a>
              </li>
            </ul>
            <a
              className="btn navbar-btn btn-ghost"
              data-target="#get-started"
              data-toggle="modal"
              href="/login"
              onClick={[Function]}
            >
              Sign Up
            </a>
            <a
              className="btn navbar-btn btn-ghost"
              data-target="#get-started"
              data-toggle="modal"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
</div>
`;
